packages/edit-site/src/components/navigation-sidebar/navigation-panel/index.js

```diff
@@ -89,18 +73,7 @@ const NavigationPanel = ( { isOpen } ) => {
 		if ( isOpen ) {
 			panelRef.current.focus();
 		}
-	}, [ templatesActiveMenu ] );
-
-	// Resets the content menu to its root whenever the navigation opens to avoid
-	// having it stuck on a sub-menu, interfering with the normal navigation behavior.
-	const prevIsOpen = usePrevious( isOpen );
-	useEffect( () => {
-		if ( contentActiveMenu !== MENU_ROOT && isOpen && ! prevIsOpen ) {
-			setContentActiveMenu( MENU_ROOT );
-		}
-	}, [ contentActiveMenu, isOpen ] );
```

Comment:
This felt like cruft after I realized the latest direction things have headed #30098.